### ISSUE TITLE
Dummy variable in the model instead of assigning continuous values to categorical variable

### ISSUE THREAD

**Issue by winterwang** [MatsuuraKentaro/RStanBook#6]
Dear Matsuura, 

I am not a native Japanese speaker, therefore please allow me to ask the question in English (thanks). 

In Page 71, where the non-aggregated data set using the weather data to predict the attendance percentages for the students, you assigned real numbers [0, 0.2, 1] to sunny days, cloudy days, and rainy days by assuming that the influence of three types of weathers might affect the logit of attendance percentages proportionally. 

I would like to ask if you can show an example of how to do this in a normal logistic regression way, ie, by creating a dummy variable for the weather variable and transfer it into a Stan Bayesian model? 

Many thanks for your advice. 

BW.

**Comment by MatsuuraKentaro** [comment]
Sorry for the late reply!

I could not understand the following parts well:
> by creating a dummy variable for the weather variable and transfer it into a Stan Bayesian model?

Does that mean you want to estimate the weather effect instead of using fixed values of 0, 0.2, 1.0? If so, the exercise (4) in that chapter is exactly that:
https://github.com/MatsuuraKentaro/RStanBook/blob/master/chap05/exercise/ex4.R
https://github.com/MatsuuraKentaro/RStanBook/blob/master/chap05/exercise/ex4.stan

**Comment by winterwang** [comment]
Yes. Thanks for pointing out about that. 

I actually did it another way as follows, but it turns out we have the same results. 
From my intuition, I will write the model as below when taking weather A as the reference: 

<a href="https://www.codecogs.com/eqnedit.php?latex=\text{logit}(q[i])&space;=&space;b_{1}&space;&plus;&space;b_{2}A_{i}&space;&plus;&space;b_{3}\text{Score}_{i}&space;&plus;&space;b_{4}\text{WeatherB}&space;&plus;&space;b_{5}\text{WeatherC}&space;\\&space;\text{where}&space;\\&space;\text{&space;WeatherB}&space;=&space;0,&space;\text{&space;WeatherC}&space;=&space;0&space;\text{&space;indicates&space;weather&space;=&space;A}&space;\\&space;\text{&space;WeatherB}&space;=&space;1,&space;\text{&space;WeatherC}&space;=&space;0&space;\text{&space;indicates&space;weather&space;=&space;B}&space;\\&space;\text{&space;WeatherB}&space;=&space;0,&space;\text{&space;WeatherC}&space;=&space;1&space;\text{&space;indicates&space;weather&space;=&space;C}&space;\\" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\text{logit}(q[i])&space;=&space;b_{1}&space;&plus;&space;b_{2}A_{i}&space;&plus;&space;b_{3}\text{Score}_{i}&space;&plus;&space;b_{4}\text{WeatherB}&space;&plus;&space;b_{5}\text{WeatherC}&space;\\&space;\text{where}&space;\\&space;\text{&space;WeatherB}&space;=&space;0,&space;\text{&space;WeatherC}&space;=&space;0&space;\text{&space;indicates&space;weather&space;=&space;A}&space;\\&space;\text{&space;WeatherB}&space;=&space;1,&space;\text{&space;WeatherC}&space;=&space;0&space;\text{&space;indicates&space;weather&space;=&space;B}&space;\\&space;\text{&space;WeatherB}&space;=&space;0,&space;\text{&space;WeatherC}&space;=&space;1&space;\text{&space;indicates&space;weather&space;=&space;C}&space;\\" title="\text{logit}(q[i]) = b_{1} + b_{2}A_{i} + b_{3}\text{Score}_{i} + b_{4}\text{WeatherB} + b_{5}\text{WeatherC} \\ \text{where} \\ \text{ WeatherB} = 0, \text{ WeatherC} = 0 \text{ indicates weather = A} \\ \text{ WeatherB} = 1, \text{ WeatherC} = 0 \text{ indicates weather = B} \\ \text{ WeatherB} = 0, \text{ WeatherC} = 1 \text{ indicates weather = C} \\" /></a>

```
#coding the weather variable into dummy variables
d <- fastDummies::dummy_cols(d, select_columns = "Weather")
```

The above code will turn weather into dummy variables shown as below: 
```
> head(d)
  PersonID A Score Weather Y Weather_B Weather_A Weather_C
1        1 0    69       B 1         1         0         0
2        1 0    69       A 1         0         1         0
3        1 0    69       C 1         0         0         1
4        1 0    69       A 1         0         1         0
5        1 0    69       B 1         1         0         0
6        1 0    69       B 1         1         0         0
```

I wrote Stan model as below `mymodel.Stan`: 

```

data {
  int I;
  int<lower=0, upper=1> A[I];
  real<lower=0, upper=1> Score[I];
  int<lower=0, upper=1> W_B[I];
  int<lower=0, upper=1> W_C[I];
  int<lower=0, upper=1> Y[I];
}

// The parameters accepted by the model. 
parameters {
  real b[5];
}

// The model to be estimated. 
model {
   for (i in 1:I)
    Y[i] ~ bernoulli_logit(b[1] + b[2]*A[i] + b[3]*Score[i] + b[4]*W_B[i] + b[5]*W_C[i]);
}
```

Then the following codes were used to run the model 

```
data <- list(I=nrow(d), A=d$A, Score=d$Score/200, 
              W_B = d$Weather_B, W_C = d$Weather_C, 
             Y=d$Y)
fit1 <- stan(file='mymodel.stan', data=data, pars=c('b'), seed=1234)
```

The results were exactly the same as the one you have provided: 

```
>fit1
Inference for Stan model: mymodel.
4 chains, each with iter=2000; warmup=1000; thin=1; 
post-warmup draws per chain=1000, total post-warmup draws=4000.

         mean se_mean   sd     2.5%      25%      50%      75%    97.5% n_eff Rhat
b[1]     0.27    0.01 0.23    -0.18     0.11     0.26     0.42     0.71  2049    1
b[2]    -0.63    0.00 0.09    -0.81    -0.69    -0.63    -0.56    -0.45  3317    1
b[3]     1.97    0.01 0.36     1.26     1.73     1.97     2.21     2.68  2308    1
b[4]    -0.38    0.00 0.11    -0.59    -0.45    -0.38    -0.31    -0.18  3378    1
b[5]    -0.49    0.00 0.12    -0.73    -0.58    -0.49    -0.41    -0.26  3194    1
lp__ -1379.61    0.04 1.55 -1383.37 -1380.43 -1379.31 -1378.48 -1377.54  1886    1

>fit
Inference for Stan model: ex4.
4 chains, each with iter=2000; warmup=1000; thin=1; 
post-warmup draws per chain=1000, total post-warmup draws=4000.

         mean se_mean   sd     2.5%      25%      50%      75%    97.5% n_eff Rhat
b[1]     0.27    0.01 0.23    -0.18     0.11     0.26     0.42     0.71  2049    1
b[2]    -0.63    0.00 0.09    -0.81    -0.69    -0.63    -0.56    -0.45  3317    1
b[3]     1.97    0.01 0.36     1.26     1.73     1.97     2.21     2.68  2308    1
bw2     -0.38    0.00 0.11    -0.59    -0.45    -0.38    -0.31    -0.18  3378    1
bw3     -0.49    0.00 0.12    -0.73    -0.58    -0.49    -0.41    -0.26  3194    1
lp__ -1379.61    0.04 1.55 -1383.37 -1380.43 -1379.31 -1378.48 -1377.54  1886    1
```


Cheers!!!
